### ISSUE TITLE
SSL Protocol error - Can't open Dropbox todo.txt

### ISSUE THREAD

**Issue by davesteele** [mpcjanssen/simpletask-android#1171]
I am no longer able to open my Dropbox todo.txt file. It always reports "Loading of todo.txt failed".

There is an exception in the log:

```
05-13 13:06:25.862 17975 18756 I FileStore: Loading file from Dropbox: /todo/todo.txt
05-13 13:06:25.862 17975 18756 I FSQ     : -> Verify token
05-13 13:06:25.863 17975 18745 I FSQ     : <- Verify token
05-13 13:06:25.943 17975 18756 E TodoList: TodoList load failed: /todo/todo.txt
05-13 13:06:25.943 17975 18756 E TodoList: com.dropbox.core.NetworkIOException: Handshake failed
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.DbxRequestUtil.startPostRaw(DbxRequestUtil.java:9)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.DbxRawClientV2$2.execute(DbxRawClientV2.java:2)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.DbxRawClientV2$2.execute(DbxRawClientV2.java:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.DbxRawClientV2.executeRetriable(DbxRawClientV2.java:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.DbxRawClientV2.downloadStyle(DbxRawClientV2.java:8)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.files.DbxUserFilesRequests.download(DbxUserFilesRequests.java:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.v2.files.DbxUserFilesRequests.download(DbxUserFilesRequests.java:4)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.remote.FileStore.loadTasksFromFile(FileStore.kt:4)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.task.TodoList.reloadaction(TodoList.kt:5)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.task.TodoList.access$reloadaction(TodoList.kt:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.task.TodoList$reload$1.invoke(TodoList.kt:9)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.task.TodoList$reload$1.invoke(TodoList.kt:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.util.ActionQueue$add$1.invoke(ActionQueues.kt:3)
05-13 13:06:25.943 17975 18756 E TodoList: 	at nl.mpcjanssen.simpletask.util.ActionQueue$add$1.invoke(ActionQueues.kt:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Async.kt:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Unknown Source:0)
05-13 13:06:25.943 17975 18756 E TodoList: 	at org.jetbrains.anko.AsyncKt$sam$java_util_concurrent_Callable$0.call(Unknown Source:2)
05-13 13:06:25.943 17975 18756 E TodoList: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
05-13 13:06:25.943 17975 18756 E TodoList: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
05-13 13:06:25.943 17975 18756 E TodoList: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
05-13 13:06:25.943 17975 18756 E TodoList: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
05-13 13:06:25.943 17975 18756 E TodoList: 	at java.lang.Thread.run(Thread.java:919)
05-13 13:06:25.943 17975 18756 E TodoList: Caused by: javax.net.ssl.SSLHandshakeException: Handshake failed
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(ConscryptFileDescriptorSocket.java:288)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.io.RealConnection.connectTls(RealConnection.java:196)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:153)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.io.RealConnection.connect(RealConnection.java:116)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:186)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:128)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:289)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:232)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:465)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:131)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:262)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.getOutputStream(DelegatingHttpsURLConnection.java:219)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:30)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.http.StandardHttpRequestor.getOutputStream(StandardHttpRequestor.java:2)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.http.StandardHttpRequestor.access$000(StandardHttpRequestor.java:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.http.StandardHttpRequestor$Uploader.<init>(StandardHttpRequestor.java:3)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.http.StandardHttpRequestor.startPost(StandardHttpRequestor.java:4)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.http.StandardHttpRequestor.startPost(StandardHttpRequestor.java:1)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.dropbox.core.DbxRequestUtil.startPostRaw(DbxRequestUtil.java:5)
05-13 13:06:25.943 17975 18756 E TodoList: 	... 21 more
05-13 13:06:25.943 17975 18756 E TodoList: Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0x74c7ea0548: Failure in SSL library, usually a protocol error
05-13 13:06:25.943 17975 18756 E TodoList: error:1000042e:SSL routines:OPENSSL_internal:TLSV1_ALERT_PROTOCOL_VERSION (external/boringssl/src/ssl/tls_record.cc:587 0x752c64b348:0x00000001)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.org.conscrypt.NativeSsl.doHandshake(NativeSsl.java:387)
05-13 13:06:25.943 17975 18756 E TodoList: 	at com.android.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(ConscryptFileDescriptorSocket.java:226)
05-13 13:06:25.943 17975 18756 E TodoList: 	... 40 more
05-13 13:06:25.943 17975 18756 I TodoList: TodoList loaded from filestore
05-13 13:06:25.943 17975 18756 I Util    : Sending file sync done changed broadcast
```

I've upgraded from 10.9.0 to 10.9.3, with no effect

**Comment by davesteele** [comment]
Note the TLSV1_ALERT_PROTOCOL_VERSION flag. Does SimpleTask need to be updated to TLS v1.3?

**Comment by davesteele** [comment]
I see this is fixed in #1159, and the latest in [master](https://mpcjanssen.nl/artifacts/master/).
